mmf_cli/run.py

```diff
@@ -108,15 +110,25 @@ def run(opts: typing.Optional[typing.List[str]] = None, predict: bool = False):
         else:
             distributed_main(0, configuration, predict)
     elif config.distributed.world_size > 1:
-        assert config.distributed.world_size <= torch.cuda.device_count()
-        port = random.randint(10000, 20000)
-        config.distributed.init_method = f"tcp://localhost:{port}"
-        config.distributed.rank = None
-        torch.multiprocessing.spawn(
-            fn=distributed_main,
-            args=(configuration, predict),
-            nprocs=config.distributed.world_size,
-        )
+        if is_xla():
+            import torch_xla.distributed.xla_multiprocessing as xmp
```

Comment:
Instead of polluting this file, it would be better to refactor these functions into a separate utils file for xla and then directly call them here for the sake of readability.